metals/src/main/scala/scala/meta/internal/metals/JsonParser.scala

```diff
@@ -0,0 +1,35 @@
+package scala.meta.internal.metals
+import com.google.gson.{Gson, JsonElement, JsonObject}
+
+import scala.reflect.{ClassTag, classTag}
+import scala.util.{Failure, Success, Try}
+
+sealed trait JsonParser {
+  implicit class Parsable(data: Any) {
+    def as[A: ClassTag]: Try[A] = {
+      data match {
+        case json: JsonElement =>
+          val targetType = classTag[A].runtimeClass.asInstanceOf[Class[A]]
+          Try(JsonParser.gson.fromJson(json, targetType))
+        case value: A =>
+          Success(value)
+        case _ =>
+          Failure(new IllegalArgumentException(s"Cannot convert [$data]"))
+      }
+    }
+  }
+
+  implicit class Serializable(data: Any) {
```

Comment:
Let's maybe do it for AnyRef.

Reply:
That would break on {de}serializing ints but it is not an important case right now, so we can do it.